Samples/WindowPlacement/cpp/inc/PlacementEx.h

```diff
@@ -0,0 +1,1596 @@
+    static bool GetPlacement(
+        HWND hwnd,
+        _Out_ PlacementEx* placement);
+
+    // Moves a window to a position stored in a PlacementEx.
+    static bool SetPlacement(HWND hwnd, PlacementEx* placement);
+
+    // Stores a placement in the registry.
+    void StoreInRegistry(
+        PCWSTR registryPath,
+        PCWSTR registryKeyName);
+
+    static void StorePlacementInRegistry(
+        HWND hwnd,
+        PCWSTR registryPath,
+        PCWSTR registryKeyName);
+
+    // FullScreen
+    //
+    // A FullScreen window is one that fills its monitor, and has some styles
+    // removed (no WS_CAPTION or WS_THICKFRAME). When entering FullScreen, the
+    // previous window position is stored. When exiting FullScreen, this position
+    // is moved to the window's monitor (which may have changed) and used to
+    // move the window to it's previous position.
+    bool EnterFullScreen(
+        HWND hwnd,
+        LONG_PTR styles = WS_CAPTION | WS_THICKFRAME);
+
+    bool ExitFullScreen(
+        HWND hwnd,
+        LONG_PTR styles = WS_CAPTION | WS_THICKFRAME);
+
+    bool IsFullScreen() const
+    {
+        return HasFlag(PlacementFlags::FullScreen);
+    }
+
+    bool ToggleFullScreen(
+        HWND hwnd,
+        LONG_PTR styles = WS_CAPTION | WS_THICKFRAME)
+    {
+        if (IsFullScreen())
+        {
+            return ExitFullScreen(hwnd, styles);
+        }
+
+        return EnterFullScreen(hwnd, styles);
+    }
+
+    // Sets the size.
+    // The provided size is logical, and scaled up by the DPI stored in the
+    // placement. The final position is adjusted to stay entirely within the
+    // work area.
+    void SetLogicalSize(SIZE size);
+
+    // Fits the placement to a different monitor.
+    // This changes the normal rect, work area, and dpi fields.
+    // By default (unless AllowPartiallyOffScreen), the final normal rect will
+    // be entirely within the bounds of the provided monitor's work area.
+    void MoveToMonitor(const MonitorData& targetMonitor);
+
+    void MoveToWindowMonitor(HWND hwnd)
+    {
+        MonitorData monitor;
+        if (MonitorData::FromWindow(hwnd, &monitor))
+        {
+            MoveToMonitor(monitor);
+        }
+    }
+
+    // Finds the monitor that best matches the position stored in the placement.
+    // This uses the device name (a string) to match monitors, and falls back to
+    // closest monitor (MonitorFromRect).
+    bool FindClosestMonitor(_Out_ MonitorData* monitor) const;
+
+    // Moves the position down/right by roughly the height of the title bar.
+    // This is done when launching two windows to the same place. Rather than
+    // have both windows in the exact same place (one covering the other),
+    // an app can choose to cascade the windows, ensuring the user can see both
+    // window's title bars. Note that this function doesn't check that scenario:
+    // it always moves the window.
+    void Cascade();
```

Comment:
Should Cascade move to the left on RTL windows?

Per offline discussion, screen coordinates do not change for RTL, so cascading might not either. It's likely worth asking an RTL expert, or at least worth calling out.